### ISSUE TITLE
some new tests fail with commit eb796c7f1 configured -Dmad -Duseithreads

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#11580]
Migrated from [rt.perl.org#97076](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=97076) (status was 'resolved')

Searchable as RT97076$

**Comment by p5pRT** [comment]
### From @tonycoz

#### Created by @tonycoz



Commit eb796c7f1 breaks several tests when built with \-Dmad \-Duseithreads

\.\./t/op/closure\.t\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.FAILED
&nbsp;    Non\-zero exit status&#8203;: 255
&nbsp;    No plan found in TAP output
\.\./ext/IPC\-Open3/t/IPC\-Open3\.t\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.FAILED
&nbsp;    Non\-zero exit status&#8203;: 255
&nbsp;    Bad plan\.  You planned 37 tests but ran 0\.
\.\./ext/Socket/t/socketpair\.t\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.FAILED
&nbsp;    Non\-zero exit status&#8203;: 255
&nbsp;    No plan found in TAP output
\.\./ext/XS\-APItest/t/swaptwostmts\.t\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.FAILED
&nbsp;    29\-32
&nbsp;    Non\-zero exit status&#8203;: 4

tony@&#8203;mars&#8203;:\.\.\./perl/t$ \./perl harness \-v op/closure\.t
op/closure\.t \.\. Bareword "WRITE" not allowed while "strict subs" in use at op/closure\.t line 412\.
Bareword "WRITE2" not allowed while "strict subs" in use at op/closure\.t line 413\.
Execution of op/closure\.t aborted due to compilation errors\.
Dubious\, test returned 255 \(wstat 65280\, 0xff00\)
No subtests run

Visible in smoke results on HP\-UX and Linux\.

Reproduced on x64 and 32\-bit Linux builds\.

Not fixed in blead as of 8367b681\.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=medium

Site configuration information for perl 5.15.1:

Configured by tony at Fri Aug 12 18:09:03 EST 2011.

Summary of my perl5 (revision 5 version 15 subversion 1) configuration:
  Commit id: eb796c7f1a47acbd996034731639c1bb76e31a19
  Platform:
    osname=linux, osvers=2.6.32-5-amd64, archname=x86_64-linux-thread-multi
    uname='linux mars 2.6.32-5-amd64 #1 smp wed jan 12 03:40:32 utc 2011 x86_64 gnulinux '
    config_args='-des -Dusedevel -Dmad -Duseithreads'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=define, usemultiplicity=define
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=define, use64bitall=define, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O2',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include'
    ccversion='', gccversion='4.4.5', gccosandvers=''
    intsize=4, longsize=8, ptrsize=8, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -fstack-protector -L/usr/local/lib'
    libpth=/usr/local/lib /lib/../lib /usr/lib/../lib /lib /usr/lib /lib64 /usr/lib64
    libs=-lnsl -lgdbm -ldl -lm -lcrypt -lutil -lpthread -lc -lgdbm_compat
    perllibs=-lnsl -ldl -lm -lcrypt -lutil -lpthread -lc
    libc=/lib/libc-2.11.2.so, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version='2.11.2'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E'
    cccdlflags='-fPIC', lddlflags='-shared -O2 -L/usr/local/lib -fstack-protector'

Locally applied patches:
    


@INC for perl 5.15.1:
    lib
    /usr/local/lib/perl5/site_perl/5.15.1/x86_64-linux-thread-multi
    /usr/local/lib/perl5/site_perl/5.15.1
    /usr/local/lib/perl5/5.15.1/x86_64-linux-thread-multi
    /usr/local/lib/perl5/5.15.1
    .


Environment for perl 5.15.1:
    HOME=/home/tony
    LANG=en_AU.UTF-8
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/tony/bin:/usr/local/bin:/usr/bin:/bin:/usr/games
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @cpansprout

On Mon Aug 15 21&#8203;:42&#8203;:10 2011\, tonyc wrote&#8203;:
> 
> This is a bug report for perl from tony@&#8203;develop\-help\.com\,
> generated with the help of perlbug 1\.39 running under perl 5\.15\.1\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please describe your issue here\]
> 
> Commit eb796c7f1 breaks several tests when built with \-Dmad
> \-Duseithreads
> 
> \.\./t/op/closure\.t\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.FAILED
>     Non\-zero exit status&#8203;: 255
>     No plan found in TAP output
> \.\./ext/IPC\-Open3/t/IPC\-Open3\.t\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.FAILED
>     Non\-zero exit status&#8203;: 255
>     Bad plan\.  You planned 37 tests but ran 0\.
> \.\./ext/Socket/t/socketpair\.t\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.FAILED
>     Non\-zero exit status&#8203;: 255
>     No plan found in TAP output
> \.\./ext/XS\-APItest/t/swaptwostmts\.t\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.FAILED
>     29\-32
>     Non\-zero exit status&#8203;: 4

I already have a fix for this\, but am waiting for the tests to complete\.
 \(I might go to bed before they finish\.\)

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @cpansprout

On Mon Aug 15 22&#8203;:32&#8203;:15 2011\, sprout wrote&#8203;:
> On Mon Aug 15 21&#8203;:42&#8203;:10 2011\, tonyc wrote&#8203;:
> > 
> > This is a bug report for perl from tony@&#8203;develop\-help\.com\,
> > generated with the help of perlbug 1\.39 running under perl 5\.15\.1\.
> > 
> > 
> > \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> > \[Please describe your issue here\]
> > 
> > Commit eb796c7f1 breaks several tests when built with \-Dmad
> > \-Duseithreads
> > 
> > \.\./t/op/closure\.t\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.FAILED
> >     Non\-zero exit status&#8203;: 255
> >     No plan found in TAP output
> > \.\./ext/IPC\-Open3/t/IPC\-Open3\.t\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.FAILED
> >     Non\-zero exit status&#8203;: 255
> >     Bad plan\.  You planned 37 tests but ran 0\.
> > \.\./ext/Socket/t/socketpair\.t\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.FAILED
> >     Non\-zero exit status&#8203;: 255
> >     No plan found in TAP output
> > \.\./ext/XS\-APItest/t/swaptwostmts\.t\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.FAILED
> >     29\-32
> >     Non\-zero exit status&#8203;: 4
> 
> I already have a fix for this\, but am waiting for the tests to complete\.
>  \(I might go to bed before they finish\.\)

Now fixed with commit 4dc304e\.

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'open' to 'resolved'
